### ISSUE TITLE
Unable to load 1.8 worlds in 1.9

### ISSUE THREAD

**Issue by copygirl** [anegostudios/VintageStory-Issues#296]
This seems to be consistent across multiple worlds so far, for me.

```
09:58:51 [Server Notification] Savegame /home/copygirl/projects/VintageStory/game/bin/Debug/Saves/modding test world.vcdbs loaded
09:58:51 [Server Notification] World size = 600000 256 600000
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: No block with code/wildcard 'game:ore-*-emerald-shale' was found (unresolved code: game:ore-*-emerald-{rock})
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: AllowedVariants for game:ore-*-emerald-shale does not match any block! Please fix
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: No block with code/wildcard 'game:ore-*-emerald-basalt' was found (unresolved code: game:ore-*-emerald-{rock})
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: AllowedVariants for game:ore-*-emerald-basalt does not match any block! Please fix
09:58:51 [Server Warning] Deposit in file game:worldgen/deposits/gem/peridot.json, no such blocks found by code/wildcard 'game:rock-peridotite'. Deposit will never spawn.
09:58:51 [Server Warning] Deposit game:worldgen/deposits/metalore/bismuthinite.json: No block with code/wildcard 'game:ore-*-bismuthinite-andesite' was found (unresolved code: game:ore-*-bismuthinite-{rock})
09:58:51 [Server Warning] Deposit game:worldgen/deposits/metalore/bismuthinite.json: AllowedVariantsByInBlock for game:ore-*-bismuthinite-andesite does not match any block! Please fix
09:58:51 [Server Warning] Deposit game:worldgen/deposits/metalore/bismuthinite.json: No block with code/wildcard 'game:ore-*-bismuthinite-granite' was found (unresolved code: game:ore-*-bismuthinite-{rock})
[...]
09:58:55 [Server Warning] Deposit game:worldgen/deposits/coal.json: No block with code/wildcard 'game:ore-anthracite-shale' was found (unresolved code: game:ore-anthracite-{rock})
09:58:55 [Server Warning] Deposit game:worldgen/deposits/coal.json: No block with code/wildcard 'game:looseores-anthracite-shale' was found (unresolved code: game:looseores-anthracite-{rock})
09:58:57 [Server Notification] Reloaded tree generators game:worldgen/treegen/dwarfbirch.json, game:worldgen/treegen/englishoak.json, game:worldgen/treegen/fir.json, game:worldgen/treegen/himalayanbirch.json, game:worldgen/treegen/japanesemaple.json, game:worldgen/treegen/kapok.json, game:worldgen/treegen/largekapok.json, game:worldgen/treegen/mountainmaple.json, game:worldgen/treegen/mountainpine.json, game:worldgen/treegen/oldenglishoak.json, game:worldgen/treegen/oldkapok.json, game:worldgen/treegen/pricklymoses.json, game:worldgen/treegen/redwoodpine.json, game:worldgen/treegen/riverbirch.json, game:worldgen/treegen/scotspine.json, game:worldgen/treegen/silverbirch.json, game:worldgen/treegen/sugarmaple.json, game:worldgen/treegen/sugarmaplesmall.json, game:worldgen/treegen/truemulga.json
09:58:57 [Server Event] Loading 7x7x8 spawn chunks...
09:58:57 [Server Event] 0% (49 in queue)
09:58:59 [Server Event] 87.8% (6 in queue)
09:58:59 [Server Notification] Server map set
09:58:59 [Server Notification] Entering runphase LoadGame
09:58:59 [Server Event] Begin save game ticking...
09:58:59 [Server Notification] Entering runphase RunGame
09:58:59 [Client Fatal] System.NullReferenceException: Object reference not set to an instance of an object
  at Vintagestory.ServerMods.DepositVariant.GetAbsAvgQuantity () [0x00001] in /home/copygirl/projects/VintageStory/game/VSSurvivalMod/Components/WorldGen/Standard/ChunkGen/5.GenDeposits/DepositVariant.cs:144
  at Vintagestory.GameContent.ProPickWorkSpace+<>c__DisplayClass4_0.<OnLoaded>b__0 () [0x000f6] in /home/copygirl/projects/VintageStory/game/VSSurvivalMod/Item/ItemProspectingPick.cs:67
  at Vintagestory.Server.ServerEventAPI.OnServerStage (Vintagestory.API.Server.EnumServerRunPhase runPhase) [0x00029] in /home/copygirl/projects/VintageStory/game/VintagestoryLib/Server/API/ServerEventAPI.cs:75
  at Vintagestory.Server.ServerSystemModHandler.OnBeginRunGame () [0x00001] in /home/copygirl/projects/VintageStory/game/VintagestoryLib/Server/Systems/ModHandler.cs:82
  at Vintagestory.Server.ServerMain.EnterRunPhase (Vintagestory.API.Server.EnumServerRunPhase runPhase) [0x000bb] in /home/copygirl/projects/VintageStory/game/VintagestoryLib/Server/ServerMain.cs:486
  at Vintagestory.Server.ServerMain.Launch () [0x00377] in /home/copygirl/projects/VintageStory/game/VintagestoryLib/Server/ServerMain.cs:429
  at Vintagestory.Client.Program.ServerThreadStart () [0x000c7] in /home/copygirl/projects/VintageStory/game/VintagestoryClient/Program.cs:324
```

**Comment by tyronx** [comment]
`09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: No block with code/wildcard 'game:ore-*-emerald-shale' was found (unresolved code: game:ore-*-emerald-{rock})
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: AllowedVariants for game:ore-*-emerald-shale does not match any block! Please fix
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: No block with code/wildcard 'game:ore-*-emerald-basalt' was found (unresolved code: game:ore-*-emerald-{rock})
09:58:51 [Server Warning] Deposit game:worldgen/deposits/gem/emerald.json: AllowedVariants for game:ore-*-emerald-basalt does not match any block! Please fix
09:58:51 [Server Warning] Deposit in file game:worldgen/deposits/gem/peridot.json, no such blocks found by code/wildcard 'game:rock-peridotite'. Deposit will never spawn.
09:58:51 [Server Warning] Deposit game:worldgen/deposits/metalore/bismuthinite.json: No block with code/wildcard 'game:ore-*-bismuthinite-andesite' was found (unresolved code: game:ore-*-bismuthinite-{rock})
09:58:51 [Server Warning] Deposit game:worldgen/deposits/metalore/bismuthinite.json: AllowedVariantsByInBlock for game:ore-*-bismuthinite-andesite does not match any block! Please fix
09:58:51 [Server Warning] Deposit game:worldgen/deposits/metalore/bismuthinite.json: No block with code/wildcard 'game:ore-*-bismuthinite-granite' was found (unresolved code: game:ore-*-bismuthinite-{rock})
[...]
09:58:55 [Server Warning] Deposit game:worldgen/deposits/coal.json: No block with code/wildcard 'game:ore-anthracite-shale' was found (unresolved code: game:ore-anthracite-{rock})
09:58:55 [Server Warning] Deposit game:worldgen/deposits/coal.json: No block with code/wildcard 'game:looseores-anthracite-shale' was found (unresolved code: game:looseores-anthracite-{rock})
`

these don't appear on my setup :o
Did you change something in the assets?

**Comment by tyronx** [comment]
oh, a 1.8 world. Can you send me that one and link it here please.

**Comment by copygirl** [comment]
Here: [modding test world.vcdbs.zip](https://github.com/anegostudios/VintageStory-Issues/files/3182927/modding.test.world.vcdbs.zip)

**Comment by tyronx** [comment]
very old bug. caused by the block id remapper derping up the first time it has stuff to re-assign, but only in some cases. Actually found and fixed the bug for v1.9.4. I hope i didn't break something else in the process.
